runelite-client/src/main/java/net/runelite/client/plugins/deathindicator/DeathIndicatorPlugin.java

```diff
@@ -0,0 +1,355 @@
+				BONES = ImageIO.read(DeathIndicatorPlugin.class.getResourceAsStream("bones.png"));
+			}
+		}
+		catch (IOException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Provides
+	DeathIndicatorConfig deathIndicatorConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DeathIndicatorConfig.class);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		client.clearHintArrow();
+
+		if (config.worldMapPointSet())
+		{
+			worldMapPointManager.removeIf(DeathWorldMapPoint.class::isInstance);
+			config.worldMapPointSet(false);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onActorDeath(ActorDeath event)
+	{
+		if (client.isInInstancedRegion())
+		{
+			return;
+		}
+
+		Actor actor = event.getActor();
+
+		if (actor instanceof NPC || actor != client.getLocalPlayer())
+		{
+			return;
+		}
+
+		config.hasDied(true);
+
+		config.hasRespawned(false);
+
+		config.deathLocationX(actor.getWorldLocation().getX());
+		config.deathLocationY(actor.getWorldLocation().getY());
+		config.deathLocationPlane(actor.getWorldLocation().getPlane());
+
+		deathLocation = new WorldPoint(config.deathLocationX(), config.deathLocationY(), config.deathLocationPlane());
+
+		config.deathWorld(client.getWorld());
+
+		if (config.showDeathHintArrow())
+		{
+			client.setHintArrow(deathLocation);
+		}
+
+		if (config.showDeathOnWorldMap())
+		{
+			addMapPoint(deathLocation);
+		}
+
+		if (deathTimer != null)
+		{
+			infoBoxManager.removeInfoBox(deathTimer);
+		}
+
+		if (config.showDeathInfoBox())
+		{
+			config.timeOfDeath(Instant.now().toString());
+			deathTimer = new Timer(60, ChronoUnit.MINUTES, BONES, plugin);
+			deathTimer.setTooltip("Died on world: " + config.deathWorld());
+			infoBoxManager.addInfoBox(deathTimer);
+		}
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getMessage().equals("Oh dear, you are dead!"))
+		{
+			config.hasRespawned(true);
+		}
+
+		if ((config.hasDied()) && (!config.timeOfDeath().equals("")))
+		{
+			if (event.getMessage().equals("Welcome to RuneScape.") && config.showDeathInfoBox())
+			{
+				if (Instant.now().isBefore(Instant.parse(config.timeOfDeath()).plus(1, ChronoUnit.HOURS)))
+				{
+					if (deathTimer != null)
+					{
+						infoBoxManager.removeInfoBox(deathTimer);
+					}
+					deathTimer = new Timer(Duration.ofHours(1).minus(Duration.between(Instant.parse(config.timeOfDeath()), Instant.now())).getSeconds(), ChronoUnit.SECONDS, BONES, plugin);
+					deathTimer.setTooltip("Died on world: " + config.deathWorld());
+					infoBoxManager.addInfoBox(deathTimer);
+				}
+			}
+		}
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (!config.hasDied())
+		{
+			return;
+		}
+
+		if (!config.hasRespawned())
+		{
+			return;
+		}
+
+		if (client.getWorld() != config.deathWorld())
```

Comment:
This should be moved to the first if statement

Reply:
You may have to just show me what you want it to look like, I can't get the plugin to work if I have the has died and has respawned on the same line or nested inside each other, and the same for the current world vs death world. Like I commented earlier, I think its because the events occur on different game ticks and if I miss the chance to continue down then the death doesn't get reccorded (it gets marked then instantly removed), but I may also just be dumb and not be doing what you want.

Reply:
Since you're just returning until all statements are true:
```java
if (!config.hasDied() || !config.hasRespawned() || client.getWorld() != config.deathWorld())
{
    return;
}
```

Reply:
Works perfectly man, thanks! I was stuck on the idea of using && rather than ||, my brain gets locked into an idea and I go blind to the obvious. I appreciate the clarification on that.